lib/fog/vsphere/requests/compute/list_child_snapshots.rb

```diff
@@ -0,0 +1,68 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Real
+        def list_child_snapshots(snapshot, opts = {})
+          s = Hash === snapshot ?
+            Snapshot.new(snapshot.update(service: self)) : snapshot
+          s.tree_node.childSnapshotList.map do |snap|
+            item = child_snapshot_info(snap, s)
+            [
+              item,
+              opts[:recursive] ? list_child_snapshots(item, opts) : nil
+            ]
+          end.flatten.compact
+        end
+
+        protected
+
+          def child_snapshot_info(snap_tree, parent_snap)
+            {
```

Comment:
Ok. You are right here. But then the `vsphere` provider need a good refactoring. It seems to be using some old standards from `fog`. Lets tackle this refactoring in another PR.